nodestream_akamai/akamai_utils/model.py

```diff
@@ -10,24 +10,26 @@ class EdgeHost:
 @dataclass(eq=True, frozen=True)
 class Origin:
     name: str
-    path: str
+    path: str | None = None
+    hostname: str | None = None
+    conditional_origin: str | None = None
 
 
 @dataclass
 class PropertyDescription:
     id: str
     name: str
-    version: str
     origins: List[Origin]
-    cloudlet_policies: List[int]
-    edge_redirector_policies: List[int]
-    edgeworker_ids: List[int]
-    siteshield_maps: List[str]
-    image_manager_policysets: List[str]
-    cp_codes: List[str]
     hostnames: List[EdgeHost]
-    ruleFormat: str
-    deeplink: str
+    version: str | None = None
+    siteshield_maps: List[str] = field(default_factory=list)
+    rule_format: str | None = None
+    image_manager_policysets: List[str] = field(default_factory=list)
+    edgeworker_ids: List[int] = field(default_factory=list)
+    edge_redirector_policies: List[int] = field(default_factory=list)
+    deeplink: str | None = None
+    cloudlet_policies: List[int] = field(default_factory=list)
+    cp_codes: List[int] = field(default_factory=list)
```

Comment:
without defaults, pycharm was warning that required constructor arguments were missing.